includes/class-wc-subscriptions-change-payment-gateway.php

```diff
@@ -331,7 +331,12 @@ public static function change_payment_method_via_pay_shortcode() {
 			// Process payment for the new method (with a $0 order total)
 			if ( wc_notice_count( 'error' ) == 0 ) {
 
-				$result = $available_gateways[ $new_payment_method ]->process_payment( $subscription->get_id() );
+				try {
+					$result = $available_gateways[ $new_payment_method ]->process_payment( $subscription->get_id() );
+				} catch ( Exception $e ) {
+					wc_add_notice( $e->getMessage(), 'error' );
+					return;
```

Comment:
I have a backwards-compatibility concern about returning before the `woocommerce_subscriptions_process_payment_for_change_method_via_pay_shortcode` filter.

https://github.com/Automattic/woocommerce-subscriptions-core/blob/1e980e5eac108bb551712bdf9952cb98ce079e5b/includes/class-wc-subscriptions-change-payment-gateway.php#L334-L344

If this filter is being used to override the `$result` to a `'success'` and we `return` before this is executed, we may introduce a bug.

Can I ask @Automattic/helix for a second opinion here?

Reply:
I might not understand your concern correctly, @Jinksi, but I see that this PR handles the case where exception is thrown by `process_payment()`, which in that case, the code execution won't get to `woocommerce_subscriptions_process_payment_for_change_method_via_pay_shortcode` either, so I think there is no backward-compatibility issue here.